keyboards/printedpad/printedpad.c

```diff
@@ -0,0 +1,55 @@
+// Copyright 2023 Noah Beidelman (@noahbei)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+#include "printedpad.h"
+
+void board_init(void) {
+  SYSCFG->CFGR1 |= SYSCFG_CFGR1_I2C1_DMA_RMP;
+}
+
+uint8_t current_layer = 0;
```

Comment:
This is already saved globally, as a bitmask of the active layers: `layer_state` (and `default_layer_state`).  You can use `get_highest_layer(layer_state)` to get the highest active layer.